### PR TITLE
[ready to review] Normalized version of filters + __repr__

### PR DESCRIPTION
**Pull request by ducha-aiki**
For better propagation it is beneficil to bound the output. Now, if one applies different filters in chain, the norm of the feature map grows very fast.
With L1 normalization of the kernel is not a such problem.
It will also simplifies things for Harris and Hessian implementation

### CODE REVIEW

File: kornia/filters/blur.py [hunk 1]
```diff
@@ -41,11 +42,21 @@ class BoxBlur(nn.Module):
     """
 
     def __init__(self, kernel_size: Tuple[int, int],
-                 border_type: str = 'reflect') -> None:
+                 border_type: str = 'reflect',
+                 normalized: bool = True) -> None:
```
Comment (edgarriba) [inline kornia/filters/blur.py]: Add arg to doc

File: kornia/filters/filter.py [hunk 2]
```diff
@@ -15,7 +16,8 @@ def compute_padding(kernel_size: Tuple[int, int]) -> List[int]:
 
 
 def filter2D(input: torch.Tensor, kernel: torch.Tensor,
-             border_type: str = 'reflect') -> torch.Tensor:
+             border_type: str = 'reflect',
+             normalized: bool = False) -> torch.Tensor:
```
Comment (edgarriba) [inline kornia/filters/filter.py]: In boxblur is true by default. Why false here?

File: kornia/filters/filter.py [hunk 3]
```diff
@@ -28,9 +30,10 @@ def filter2D(input: torch.Tensor, kernel: torch.Tensor,
           :math:`(B, C, H, W)`.
         kernel (torch.Tensor): the kernel to be convolved with the input
           tensor. The kernel shape must be :math:`(B, kH, kW)`.
-        borde_type (str): the padding mode to be applied before convolving.
+        border_type (str): the padding mode to be applied before convolving.
           The expected modes are: ``'constant'``, ``'reflect'``,
           ``'replicate'`` or ``'circular'``. Default: ``'reflect'``.
+        normalized (bool): If True, kernel will be L1 normalized
```
Comment (edgarriba) [inline kornia/filters/filter.py]: End with point  '.'

File: kornia/filters/kernels.py [hunk 4]
```diff
@@ -4,6 +4,13 @@
 import torch.nn as nn
 
 
+def normalize_kernel2d(input: torch.Tensor) -> torch.Tensor:
+    r"""Normalizes both derivative and smoothing kernel.
+    """
```
Comment (edgarriba) [inline kornia/filters/kernels.py]: No asserts checking types and shapes ?

File: kornia/filters/laplacian.py [hunk 5]
```diff
@@ -34,12 +35,22 @@ class Laplacian(nn.Module):
     """
 
     def __init__(self,
-                 kernel_size: int, border_type: str = 'reflect') -> None:
+                 kernel_size: int, border_type: str = 'reflect',
+                 normalized: bool = True) -> None:
         super(Laplacian, self).__init__()
         self.kernel_size: int = kernel_size
         self.border_type: str = border_type
+        self.normalized: bool = normalized
```
Comment (edgarriba) [inline kornia/filters/laplacian.py]: Add doc

File: kornia/filters/sobel.py [hunk 6]
```diff
@@ -21,11 +22,25 @@ class SpatialGradient(nn.Module):
         >>> output = kornia.filters.SpatialGradient()(input)  # 1x3x2x4x4
     """
 
-    def __init__(self, mode: str = 'sobel', order: int = 1) -> None:
+    def __init__(self,
+                 mode: str = 'sobel',
+                 order: int = 1,
+                 normalized: bool = True) -> None:
         super(SpatialGradient, self).__init__()
+        self.normalized: bool = normalized
```
Comment (edgarriba) [inline kornia/filters/sobel.py]: Add params to docs

File: kornia/filters/sobel.py [hunk 7]
```diff
@@ -88,18 +110,19 @@ def forward(self, input: torch.Tensor) -> torch.Tensor:  # type: ignore
...
 
 
-def sobel(input: torch.Tensor) -> torch.Tensor:
+def sobel(input: torch.Tensor, normalized: bool = True) -> torch.Tensor:
     r"""Computes the Sobel operator and returns the magnitude per channel.
 
     See :class:`~kornia.filters.Sobel` for details.
     """
-    return Sobel()(input)
+    return Sobel(normalized=normalized)(input)
```
Comment (edgarriba) [inline kornia/filters/sobel.py]: why explicit param ?

File: setup.cfg [hunk 8]
```diff
@@ -14,3 +14,13 @@ max-line-length = 80
 max-line-length = 80
 ignore = F401,E402,F403,F811
 exclude = venv
+
+[report]
+exclude_lines =
+    pragma: no cover
+    def __repr__
```
Comment (edgarriba) [inline setup.cfg]: Not sure it makes effect

File: kornia/filters/filter.py (hunk 2 above)
Comment (ducha-aiki) [reply kornia/filters/filter.py]: Because everywhere else, is it us who set defaults. 
For filter2D, I believe, user would provide the exactly filter she wants to convolve with. I think, don`t change the filter be default is good thing

File: setup.cfg (hunk 8 above)
Comment (ducha-aiki) [reply setup.cfg]: I copy-pasted the thing from the docs :)
https://coverage.readthedocs.io/en/coverage-4.2/excluding.html

but yes, it does not work somehow.

Comment (edgarriba) [reply setup.cfg]: Or just add a quick test with default params

File: kornia/filters/kernels.py [hunk 9]
```diff
@@ -7,6 +7,9 @@
 def normalize_kernel2d(input: torch.Tensor) -> torch.Tensor:
     r"""Normalizes both derivative and smoothing kernel.
     """
+    if len(input.size()) < 2:
+        raise TypeError("input should be at least 2D tensor. Got {}"
+                        .format(input.size()))
     norm: torch.Tensor = input.abs().sum(dim=-1).sum(dim=-1)
     return input / (norm.unsqueeze(-1).unsqueeze(-1))
```
Comment (edgarriba) [inline kornia/filters/kernels.py]: should we prevent from zero division ?
